lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -431,14 +430,10 @@ def deprecate(self, msg, version=None):
         self.module.deprecate(msg, version)
 
     def log(self, msg, pretty_print=False):
-        pass
-        # Use only during module development
-        # if self.debug:
-        #     log_file = open('azure_rm.log', 'a')
-        #     if pretty_print:
-        #         log_file.write(json.dumps(msg, indent=4, sort_keys=True))
-        #     else:
-        #         log_file.write(msg + u'\n')
+        if pretty_print:
+            self.module.debug(json.dumps(msg, indent=4, sort_keys=True))
```

Comment:
How about the message is not a json but pretty_print is True?

Reply:
It will crash as per normal python. But given the entire point of the `pretty_print` option appears to be so that it is json_dump'd that would be a PBUAK problem and it kinda should blow up

Reply:
Can we detect if this is not a json and fallback to print it directly

Reply:
Only by catching the exception and then falling back to the same line as if `pretty_print` wasn't passed as true to start with (which by definition would make it a module bug)

Reply:
Personally I wouldn't even have the option - it should be the job of the module to format appropriately rather than the common module trying to guess the right format. But given it's there I need to make it work...